lib/core/managementClient.ts

```diff
@@ -532,6 +556,209 @@ export class ManagementClient extends LinkEntity {
     }
   }
 
+
+  /**
+   * Get all the rules on the Subscription.
+   * @returns Promise<RuleDescription> A list of rules.
+   */
+  async getRules(): Promise<RuleDescription[]> {
+    try {
+      const request: AmqpMessage = {
+        body: {
+          top: types.wrap_int(max32BitNumber),
+          skip: types.wrap_int(0)
+        },
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.enumerateRules
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+
+      log.mgmt("[%s] Get rules request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      const response = await this._mgmtReqResLink!.sendRequest(request);
+      if (response.application_properties!.statusCode === 204 || !response.body || !Array.isArray(response.body.rules)) {
+        return [];
+      }
+
+      // Reference: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#response-11
+      const result: { "rule-description": Typed }[] = response.body.rules || [];
+      const rules: RuleDescription[] = [];
+      result.forEach((x) => {
+        const ruleDescriptor = x['rule-description'];
+        if (!Array.isArray(ruleDescriptor.value) || ruleDescriptor.value.length < 3) {
```

Comment:
Why 3? Would you mind adding a comment to describe what that means?

Reply:
The comment with the reference link a few lines above describes the schema of the response object. So I didnt add a specific comment for this.

We use the first 3 elements of the array for filter, action and name of the rule.

They could change the schema in the future to add more details, so I chose to go with < 3 instead of != 3

Reply:
Got it. I would still add this comment right here in the code so that people don't have to read and understand the docs at the link you specified.

Reply:
Done